builder/builder.go

```diff
@@ -73,22 +82,33 @@ type Builder struct {
 
 // BuilderArgs is a struct that contains all the arguments needed to create a new Builder
 type BuilderArgs struct {
-	sk                          *bls.SecretKey
-	ds                          flashbotsextra.IDatabaseService
-	relay                       IRelay
-	builderSigningDomain        boostTypes.Domain
-	eth                         IEthereumService
-	dryRun                      bool
-	ignoreLatePayloadAttributes bool
-	validator                   *blockvalidation.BlockValidationAPI
-	beaconClient                IBeaconClient
+	sk                           *bls.SecretKey
+	ds                           flashbotsextra.IDatabaseService
+	relay                        IRelay
+	builderSigningDomain         boostTypes.Domain
+	builderBlockResubmitInterval time.Duration
+	eth                          IEthereumService
+	dryRun                       bool
+	ignoreLatePayloadAttributes  bool
+	validator                    *blockvalidation.BlockValidationAPI
+	beaconClient                 IBeaconClient
+
+	limiter *rate.Limiter
 }
 
 func NewBuilder(args BuilderArgs) *Builder {
 	pkBytes := bls.PublicKeyFromSecretKey(args.sk).Compress()
 	pk := boostTypes.PublicKey{}
 	pk.FromSlice(pkBytes)
 
+	if args.limiter == nil {
+		args.limiter = rate.NewLimiter(rate.Every(RateLimitIntervalDefault), RateLimitBurstDefault)
+	}
+
+	if args.builderBlockResubmitInterval == 0 {
```

Comment:
I wonder if this shouldn't be the caller's responsibility (same for the above)

Reply:
> I wonder if this shouldn't be the caller's responsibility (same for the above)

I have it in both places. In general I agree with you re: caller's responsibility. However I would want to return an error for invalid inputs but the function signature would need to be updated. I didn't want to do that work in this PR - happy to discuss

Reply:
> I'd make it the responsibility of the caller still, since you don't really validate the value

For now what's wrong with having it in both places? I'm worried that without the defensive check in the receiver, we can easily have a resubmit interval of 0. My preference would rather be to refactor the function signature in the future such that a 0 value returns an error.